jsonschema-maven-plugin/src/main/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojo.java

```diff
@@ -204,10 +204,10 @@ private File getSchemaFile() {
     private SchemaGenerator createGenerator(ObjectMapper mapper) throws MojoExecutionException {
         // Start with the generator builder
         SchemaGeneratorConfigBuilder configBuilder =
-                new SchemaGeneratorConfigBuilder(mapper, getSchemaVersion(), OptionPreset.PLAIN_JSON);
+                new SchemaGeneratorConfigBuilder(mapper, getSchemaVersion(), getOptionPreset());
 
         // Add options when required
-        if (options.length == 0) {
+        if (options == null) {
             setDefaultOptions(configBuilder);
         } else {
```

Comment:
Let's also avoid adding the `Option.DEFINITIONS_FOR_ALL_OBJECTS` by default in addition to whatever preset is being selected.
```suggestion
        if (options != null) {
```

Reply:
Fixed